rust/typeql_lang.rs

```diff
@@ -20,5 +20,70 @@
  *
  */
 
+#![allow(dead_code)]
+
+extern crate core;
+
+use std::cell::RefCell;
+use std::convert::Into;
+use std::rc::Rc;
+
+use typeql_grammar::typeqlrustlexer::TypeQLRustLexer;
+use typeql_grammar::typeqlrustparser::*;
+
+use antlr_rust::common_token_stream::CommonTokenStream;
+use antlr_rust::{InputStream, Parser as ANTLRParser};
+
+pub mod common;
+pub mod parser;
+pub mod pattern;
 pub mod query;
-pub mod parser;
\ No newline at end of file
+
+#[macro_use]
+mod util;
+
+use crate::parser::{error_listener::ErrorListener, syntax_error::SyntaxError, visit_eof_query};
+use pattern::*;
+use query::*;
+
+pub fn parse_query(typeql_query: &str) -> Result<Query, String> {
+    parse_eof_query(typeql_query)
+}
+
+pub fn typeql_match(pattern: impl Into<Conjunction>) -> Query {
+    Query::Match(TypeQLMatch::new(pattern.into()))
+}
+
+pub fn var(var: impl Into<UnboundVariable>) -> UnboundVariable {
+    var.into()
+}
+
+pub fn type_(name: impl Into<String>) -> BoundVariable {
+    UnboundVariable::hidden().type_(name.into())
+}
+
+pub fn rel<T: Into<RolePlayerConstraint>>(value: T) -> BoundVariable {
+    UnboundVariable::hidden().rel(value)
+}
+
+pub fn parse_eof_query(query_string: &str) -> Result<Query, String> {
```

Comment:
Multiple syntax errors don;t really fit into the `ErrorMessage` abstraction. Needs work. For now, return errors as Strings.